docs/topics/gradle-best-practices.md

```diff
@@ -0,0 +1,141 @@
+[//]: # (title: Gradle best practices)
+
+Getting the best out of Gradle is essential to help you spend less time managing and waiting for builds, and more time 
+coding. Here we provide a set of best practices split into two key areas: **organizing** and **optimizing** your projects.
+
+## Organize
+
+This section focuses on structuring your Gradle projects to improve clarity, maintainability, and scalability.
+
+### Use a version catalog
+
+Use a version catalog in a `libs.versions.toml` file to centralize dependency management, enabling you to define and 
+reuse versions, libraries, and plugins consistently across projects.
+
+```none
+[versions]
+kotlinxCoroutines = "%coroutinesVersion%"
+
+[libraries]
+kotlinxCoroutines = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-core", version.ref = "kotlinxCoroutines" }
+```
+
+With the following dependency added to your `build.gradle.kts` file:
+
+```none
+dependencies {
+    implementation(libs.kotlinxCoroutines)
+}
+```
+
+Learn more in Gradle's documentation about [Dependency management basics](https://docs.gradle.org/current/userguide/dependency_management_basics.html#version_catalog).
+
+### Use Kotlin DSL
+
+Use Kotlin DSL instead of the traditional Groovy DSL. You avoid learning another language and gain the benefits of strict
+typing. Strict typing lets IDEs provide better support for refactoring and auto-completion, making development more efficient.
+
+Find more information in [Gradle’s Kotlin DSL primer](https://docs.gradle.org/current/userguide/kotlin_dsl.html).
+
+Read Gradle's [blog](https://blog.gradle.org/kotlin-dsl-is-now-the-default-for-new-gradle-builds) about Kotlin DSL becoming
+the default for Gradle builds.
+
+### Use convention plugins
+
+<primary-label ref="advanced"/>
+
+Use convention plugins to encapsulate and reuse common build logic across multiple build files. Moving shared configurations
+into a plugin helps simplify and modularize your build scripts.
+
+Although the initial setup may be time-consuming, you will find it easy to maintain and add new build logic once you complete it.
+
+Learn more in Gradle's documentation about [Convention plugins](https://docs.gradle.org/current/userguide/custom_plugins.html#sec:convention_plugins).
```

Comment:
It looks like only the first paragraph is highlighted as 'advanced', but should all these three paragraphs be highlighted?

<img width="730" alt="image" src="https://github.com/user-attachments/assets/76757d77-b42f-4de5-986a-b7932c8b726e" />

Reply:
That's the style of labels in writerside: https://www.jetbrains.com/help/writerside/topic-and-chapter-labels.html#insert-labels The only alternative atm is to add the label per paragraph but that would look awful.